src/lein_monolith/task/each.clj

```diff
@@ -313,16 +314,26 @@
   (mapv (comp (partial run-task! ctx) second) targets))
 
 
+(defn- resolve-tasks
+  "Perform an initial resolution of the task to prevent metadata-related
+  arglist errors when namespaces are loaded in parallel."
+  [task+args]
+  (let [task (first task+args)]
+    (lein/resolve-task task)
+    ;; A 'do' task pulls in other tasks, so also resolve them.
+    (when (= "do" task)
```

Comment:
I think you should go ahead and add `with-profile` as well. Should we resolve those recursively in those cases?

Reply:
Ah, good call on recursion.